### PR TITLE
Convert RSVP to native Promise throughout codebase.

### PR DESCRIPTION
**Pull request by mbleigh**
Now that we are Node 4 and up, we can stop relying on RSVP for basic Promise implementation. This replaces RSVP everywhere and removes it as a dependency. It also updates `portfinder` to a version that has a promise-based implementation.

### CODE REVIEW

File: commands/help.js [hunk 1]
```diff
@@ -28,5 +28,5 @@ module.exports = new Command("help [command]")
       logger.info();
     }
 
-    RSVP.resolve();
+    Promise.resolve();
```
Comment (laurenzlong) [inline commands/help.js]: does this need to be `return Promise.resolve()`?

File: lib/command.js [hunk 2]
```diff
@@ -111,7 +110,7 @@ Command.prototype.register = function(client) {
         var projectId = getProjectId(options, true);
         var preppedMessage = chalk.stripColor(err.message || "").replace(projectId, "<namespace>");
 
-        RSVP.all([
+        Promise.all([
```
Comment (laurenzlong) [inline lib/command.js]: does this need to get returned?

**Comment by laurenzlong** [comment]
Node 4 also supports `let`, `const` and `class`. We don't have to do it now, but we may want to consider replacing all the vars in our code at some point

File: commands/help.js (hunk 1 above)
Comment (mbleigh) [reply commands/help.js]: Done.

File: lib/command.js (hunk 2 above)
Comment (mbleigh) [reply lib/command.js]: Done.

**Comment by mbleigh** [comment]
Addressed.

**Pull request merged**